doc/designs/xof.md

```diff
@@ -78,13 +78,17 @@ EVP_DigestSqueeze(ctx, out, outlen).
 ##### Pros
 
   - Seems like a better name.
+  - The existing function does not change, so it is not affected by logic that needs to
+    run for the multi squeeze case.
+  - The behaviour of the existing API is the same.
+  - At least one other toolkit uses this approach.
 
 ##### Cons
 
   - Adds an extra API.
   - The interaction between the 2 API's needs to be clearly documented.
-     - A call to EVP_DigestSqueeze() after EVP_DigestFinalXOF() would fail.
-     - A call to EVP_DigestFinalXOF() after the EVP_DigestSqueeze() would fail? Is this confusing.
+  - A call to EVP_DigestSqueeze() after EVP_DigestFinalXOF() would fail.
```

Comment:
Not sure why this needs to fail

Reply:
Because FInalXOF means we are doing a one shot output.. (i.e. The code works as it already did before for this case).
Squeeze is the interface if you want to do one or more output calls..
With this definitiion FinalXOF followed by Squeeze should fail... 
This also keeps the code almost unchanged for the old case.

Reply:
OK, if it means final() not needing changes but it would need them otherwise, then that makes sense.